src/main/java/org/folio/rest/impl/AbstractInstanceRecordsAPI.java

```diff
@@ -99,7 +99,17 @@ protected void fetchRecordsByQuery(String sql, Supplier<Tuple> paramsSupplier, R
 
         RowStream<Row> rowStream = ar.result();
         rowStream
-        .exceptionHandler(e -> respondWithError(response, e, asyncResultHandler))
+        .exceptionHandler(e -> {
+          respondWithError(response, e, asyncResultHandler);
+          //database connections must be explicitly closed when an error is thrown 
+          //or it will be left in an open aborted state.
+          postgresClient.endTx(tx, h -> {
+            if (h.failed()) {
+              log.error("Unable to close database connection: " + h.cause().getMessage());
+              return;
+            }
```

Comment:
Could this be extracted into a separate method / class that could be reused (in other places where this could occur) and tested independently (to improve the coverage), maybe by executing some SQL known to abort to the connection?

Reply:
I'll try.